src/Workspaces/Core/Portable/Diagnostics/Extensions.cs

```diff
@@ -81,14 +82,33 @@ public static string GetAnalyzerId(this DiagnosticAnalyzer analyzer)
 
     private static string GetAssemblyQualifiedName(Type type)
     {
-        // AnalyzerFileReference now includes things like versions, public key as part of its identity. 
+        // AnalyzerFileReference now includes things like versions, public key as part of its identity.
         // so we need to consider them.
         return RoslynImmutableInterlocked.GetOrAdd(
             ref s_typeToAssemblyQualifiedName,
             type,
             static type => type.AssemblyQualifiedName ?? throw ExceptionUtilities.UnexpectedValue(type));
     }
 
+    private static readonly ImmutableHashSet<string> s_featuresAnalyzerFileNames = new[] {
+        "Microsoft.CodeAnalysis.Features.dll",
+        "Microsoft.CodeAnalysis.CSharp.Features.dll",
+        "Microsoft.CodeAnalysis.VisualBasic.Features.dll",
+    }.ToImmutableHashSet(StringComparer.OrdinalIgnoreCase);
+
+    private static ImmutableSegmentedDictionary<string, bool> s_analyzerFullPathToIsFeatureAnalyzer = ImmutableSegmentedDictionary<string, bool>.Empty;
```

Comment:
Do we need to cache this? It's just simple comparison of 3 strings.

Reply:
You can also do:

```
bool IsFeaturesAnalyzer(this AnalyzerReference reference)
{
  var fileNameSpan = reference.FullPath.AsSpan(FileNameUtilities.IndexOfFileName(reference.FullPath));
  return 
    fileNameSpan.Equals("Microsoft.CodeAnalysis.Features.dll", StringComparison.OrdinalIgnoreCase) ||
    fileNameSpan.Equals("Microsoft.CodeAnalysis.CSharp.Features.dll", StringComparison.OrdinalIgnoreCase) ||
    fileNameSpan.Equals("Microsoft.CodeAnalysis.VisualBasic.Features.dll", StringComparison.OrdinalIgnoreCase);
}
```

Reply:
agreed.  dictionaries fo rthis seem unnecessary.